Финальный проект 7 спринта

AlexeyLandik opened the pull request.

AlexeyLandik committed:
Финальный проект 7 спринта

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,16 @@
+import pytest
+
+
+@pytest.fixture()
+def order_data():
+    order_data = {
+        "firstName": "Naruto",
+        "lastName": "Uchiha",
+        "address": "Konoha, 142 apt.",
+        "metroStation": 4,
+        "phone": "+7 800 355 35 35",
+        "rentTime": 5,
+        "deliveryDate": "2020-06-06",
+        "comment": "Saske, come back to Konoha"
+    }
+    return order_data
```
Нужно исправить: фикстуры не занимаются прокидыванием данных в тест, они выполняют логику предусловий\постусловий. Данные из data нужно прости импортировать куда необходимо

AlexeyLandik replied:
Согласен. Написал новую фикстуру.

AlexKitIt commented on tests/helpers.py:
```diff
@@ -0,0 +1,62 @@
+    @allure.step('Регистрация курьера и возврат ответа сервера')
+    def register_courier(self, user_data: dict):
+        response = requests.post(data.CREATE_COURIER, data=user_data)
+        print(response.text)
```
Нужно исправить: принтов быть не должно

AlexeyLandik replied:
Убрал

AlexKitIt commented on tests/test_courier.py:
```diff
@@ -0,0 +1,60 @@
+        new_courier_data = courier.courier_datas_creation(10)
+        new_courier_data.pop('password')
+        response = requests.post(data.CREATE_COURIER, data=new_courier_data)
+        assert response.status_code == 400 and response.json()['message'] == ("Недостаточно данных для создания "
```
Можно лучше: тексты ответов стоит хранить снаружи в data модулях для поддержания и актуализации

AlexeyLandik replied:
Принял. Учту в дальнейшей работе.

AlexKitIt commented on tests/test_courier.py:
```diff
@@ -0,0 +1,60 @@
+        courier = Courier()
+        new_courier_data_1 = courier.courier_datas_creation(10)
+        requests.post(data.CREATE_COURIER, data=new_courier_data_1)
+        new_courier_data_2 = courier.courier_datas_creation(10)
```
А что мешает передать new_courier_data_1 во второй запрос?

AlexeyLandik replied:
Смысл был показать, что второй пользователь дублирует только логин первого пользователя.
В актуальной версии предаю то же самое и во второй запрос.

AlexeyLandik committed:
Финальный проект 7 спринта. Исправленный

AlexKitIt commented on data.py:
```diff
@@ -2,5 +2,6 @@
 
 CREATE_COURIER = BASE_URL + '/api/v1/courier'  # Ручка для создания курьера
 LOGIN_COURIER = BASE_URL + '/api/v1/courier/login'  # Ручка для авторизации курьера
+DELETE_COURIER = BASE_URL + '/api/v1/courier/'  # Ручка для удаления курьера
```
Так одинаковый же url

AlexeyLandik replied:
Убрал задвоение константы

AlexKitIt commented on tests/conftest.py:
```diff
@@ -1,4 +1,5 @@
+from helpers_files.helpers import Courier
 
 
 @pytest.fixture()
```
Нужно исправить: повторный коммент: фикстуры такой быть не должно, класс описывается в data и импортируется в тесты заказа

AlexeyLandik replied:
- убрал фикстуру, кт прокидывала данные в тест
- описал константы в data.py через класс и импортировал в тесты

AlexeyLandik committed:
Финальный проект 7 спринта. Второе исправление.
1) Убрал задвоение константы (Так одинаковый же url)
2) Повторное замечание по фикстуре:
- убрал фикстуру, кт прокидывала данные в тест
- описал константы в data.py через класс и импортировал в тесты

AlexeyLandik committed:
Финальный проект 7 спринта. Второе исправление.
1) Убрал задвоение константы (Так одинаковый же url)
2) Повторное замечание по фикстуре:
- убрал фикстуру, кт прокидывала данные в тест
- описал константы в data.py через класс и импортировал в тесты
